frontend/packages/console-shared/src/components/dynamic-form/utils.ts

```diff
@@ -175,15 +210,21 @@ export const getJSONSchemaOrder = (jsonSchema, uiSchema) => {
     }
 
     const uiOrder = Immutable.Set(propertyNames)
-      .sortBy((propertyName) => getJSONSchemaPropertySortWeight(propertyName, jsonSchema, uiSchema))
+      .sortBy((property) =>
+        getJSONSchemaPropertySortWeight(property, jsonSchema, uiSchema, currentPath ?? []),
+      )
       .toJS();
 
     return {
       ...(uiOrder.length > 1 && { 'ui:order': uiOrder }),
       ..._.reduce(
         jsonSchema?.properties ?? {},
         (orderAccumulator, propertySchema, propertyName) => {
-          const descendantOrder = getJSONSchemaOrder(propertySchema, uiSchema?.[propertyName]);
+          const descendantOrder = getJSONSchemaOrder(
+            propertySchema as JSONSchema6,
```

Comment:
Can you simply declare the type above in the function definition like

```ts
(orderAccumulator, propertySchema: JSONSchema6, propertyName) => {
```

This is more type safe than using `as`